packages/api/src/mappings/google/activity.ts

```diff
@@ -99,22 +208,48 @@ export const mapToActivity = (
     });
   }
 
-  if (googleSessions) {
-    const activitySessions = googleSessions.session.filter(
-      session => session.activityType !== sessionSleepType
-    );
-
-    const activityLogs = activitySessions.map(session => {
-      return {
-        metadata,
-        name: GoogleActivityTypes[session.activityType],
-        start_time: dayjs(Number(session.startTimeMillis)).toISOString(),
-        end_time: dayjs(Number(session.endTimeMillis)).toISOString(),
-      };
-    });
+  return activity;
+};
 
-    activity.activity_logs = activityLogs;
+function nanoTimeString(startTime: string | undefined): number {
+  if (startTime) {
+    const dateObject = new Date(startTime);
+    const timeInNanoseconds = dateObject.getTime() * 1e6;
+    return timeInNanoseconds;
   }
+  return 0;
+}
 
-  return activity;
-};
+function matchActivityTime(
+  activityStartTime: string | undefined,
+  activityEndTime: string | undefined,
+  startTimeNanos: number
+): boolean {
+  return (
+    nanoTimeString(activityStartTime) <= startTimeNanos &&
+    startTimeNanos < nanoTimeString(activityEndTime)
+  );
+}
+
+function formatNumber(num: number): number {
+  return parseInt((num * 100).toFixed(2)) / 100;
+}
+
+function calculateAvgSpeed(activitySpeedTimeMap: { speed: number; totalSeconds: number }[]) {
+  let totalDistance = 0;
+  let totalTime = 0;
+
+  activitySpeedTimeMap.forEach(d => {
+    totalDistance += (d.speed * d.totalSeconds) / 3600;
+    totalTime += d.totalSeconds;
+  });
+
+  return formatNumber((totalDistance / totalTime) * 3600);
```

Comment:
`3600` is a magic number that's not useful, I can only assume this is "seconds in an hour"

let's use our convert lib here?

ie `convert(1).from("h").to("s")` is much better